SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp

```diff
@@ -0,0 +1,370 @@
+        m_display_order.emplace_back("Losses");
+        m_display_order.emplace_back("Errors", true);
+    }
+    std::atomic<uint64_t>& tournaments;
+    std::atomic<uint64_t>& battles;
+    std::atomic<uint64_t>& losses;
+    std::atomic<uint64_t>& errors;
+};
+std::unique_ptr<StatsTracker> TournamentFarmer_Descriptor::make_stats() const {
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+TournamentFarmer::TournamentFarmer()
+    : NUM_ROUNDS(
+          "<b>Number of Tournaments to run:",
+          LockWhileRunning::UNLOCKED,
+          100, 0
+          )
+    , TRY_TO_TERASTILLIZE(
+          "<b>Use Terastillization:</b><br>Tera at the start of battle. Will take longer but may be worth the attack boost.",
+          LockWhileRunning::UNLOCKED,
+          false
+          )
+    , SAVE_NUM_ROUNDS(
+          "<b>Save every this many tournaments:</b><br>(zero disables saving)",
+          LockWhileRunning::UNLOCKED,
+          0, 0
+          )
+    , GO_HOME_WHEN_DONE(false)
+    , LANGUAGE(
+          "<b>Game Language:</b><br>The language is needed to read the prizes.",
+          TournamentPrizeNameReader::instance().languages(),
+          LockWhileRunning::LOCKED,
+          true
+          )
+    , TARGET_ITEMS("<b>Items:</b>")
+    , NOTIFICATION_PRIZE_MATCH("Matching Prize", true, false, ImageAttachmentMode::JPG, { "Notifs" })
+    , NOTIFICATION_STATUS_UPDATE("Status Update", true, false, std::chrono::seconds(3600))
+    , NOTIFICATIONS({
+        &NOTIFICATION_PRIZE_MATCH,
+        &NOTIFICATION_STATUS_UPDATE,
+        &NOTIFICATION_PROGRAM_FINISH,
+        &NOTIFICATION_ERROR_FATAL,
+        })
+{
+    PA_ADD_OPTION(NUM_ROUNDS);
+    PA_ADD_OPTION(TRY_TO_TERASTILLIZE);
+    PA_ADD_OPTION(SAVE_NUM_ROUNDS);
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(LANGUAGE);
+    PA_ADD_OPTION(TARGET_ITEMS);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+void TournamentFarmer::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
+    assert_16_9_720p_min(env.logger(), env.console);
+    TournamentFarmer_Descriptor::Stats& stats = env.current_stats<TournamentFarmer_Descriptor::Stats>();
+
+    /*
+    Preconditions:
+    Last Pokemon Center visited is Mesagzoa West
+    Sylveon only farming build - ideally with fairy tera
+    stand in front of tournament entry man
+    Ride legendary is not the solo pokemon (in case of loss)
+
+    Possible improvements to make:
+    money tracking? notifs tend to backlog.
+    find prize sprites - some code is there, just disabled
+    find translations for tera shards
+    other languages: make sure "bottle cap" isn't misread as "bottle of PP Up"
+    */
+
+    for(uint32_t c = 0; c < NUM_ROUNDS; c++) {
+        //Initiate dialog then mash until first battle starts
+        AdvanceDialogWatcher advance_detector(COLOR_YELLOW);
+        pbf_press_button(context, BUTTON_A, 10, 50);
+        pbf_press_button(context, BUTTON_A, 10, 50); //sometimes the input drops
```

Comment:
is it only the first time ? or every time (the first button press is usually lost connecting the controller)